### PR TITLE
Add Cache-Control header to search

### PR DESCRIPTION
**Pull request by gmethvin**
This is useful so if we're doing instant search we don't request the same search multiple times.

### CODE REVIEW

**Comment by eishay** [comment]
what does it mean? caching search results in the client? we don't want it - not even for 30 seconds

**Comment by gmethvin** [comment]
If we're doing an instant search and type backspace during the search, we'd rather not have the browser make another request to get the same results. Setting the Expires time tells the browser that it can cache the result and use the response it just got.

**Comment by ymatsuda** [comment]
I think it is a great idea, but 30 seconds looks too long to me.

File: shoebox/app/com/keepit/controllers/ext/ExtSearchController.scala [hunk 1]
```diff
@@ -139,14 +139,14 @@ class ExtSearchController @Inject() (
...
-      		"\n More details at: \n" + link + "\n" + searchDetails + "\n"
+        "\n More details at: \n" + link + "\n" + searchDetails + "\n"
       healthcheckPlugin.addError(HealthcheckError(
         error = Some(new SearchTimeExceedsLimit(timeLimit, total)),
         errorMessage = Some(msg),
         callType = SEARCH))
     }
 
-    Ok(RPS.resSerializer.writes(res)).as(ContentTypes.JSON)
+    Ok(Json.toJson(res)).withHeaders("Expires" -> clock.now().plusSeconds(30).toHttpHeaderString)
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/ext/ExtSearchController.scala]: shorter, clearer over the wire, more foolproof (no use of a clock in our code) and [equivalent](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9.3):
`"Cache-Control" -> "private, max-age=30"`

**Comment by gmethvin** [comment]
15 seconds?

**Comment by gmethvin** [comment]
Oh and also Jared suggested `Cache-Control: private, max-age=30` since it's more clear. I tend to agree.

**Comment by eishay** [comment]
Yasuhiro's call

**Comment by ymatsuda** [comment]
I am not sure. I think it is also related to a user experience. How fresh should the result be?

**Comment by gmethvin** [comment]
If the user hasn't made a particular query in the last 30 seconds, the results should be as fresh as before.

I guess this might create a problem if the user searches, then keeps something, then searches immediately after expecting the keep to be there. I don't think this is that common though.

**Comment by yingjieMiao** [comment]
That may be common for new users though.

**Comment by gmethvin** [comment]
Setting it to 10 seconds

**Pull request merged**